### PR TITLE
[tools] Prevent from checking whether google-services.json is present in bare-expo app.

### PR DESCRIPTION
**Pull request by mczernek**
# Why

Currently, google-services.json is copied to `apps/bare-expo/android/app`. Then, i.e. when creating shellapp, its existence is verified. Shellapp has no notion of bare-expo app, which causes the check to fail.

# How

Divided `android-paths.json` entries into `paths` and `generateOnly`, the latter of which are not being verifier.

# Test Plan

Run `et android-generate-dynamic-macros` on your machine, delete `google-services.json` from bare-expo app, and run `./tools-public/check-dynamic-macros-android.sh`. If it passes, everything is all right.

### CODE REVIEW

File: update-dependencies [hunk 1]
```diff
@@ -20,6 +20,7 @@ bin/nix-binstubs \
   fastlane \
   git-crypt \
   git-lfs \
+  js \
```
Comment (sjchmiela) [inline update-dependencies]:
```suggestion
  jq \
```

File: tools/expotools/src/dynamic-macros/generateDynamicMacros.ts [hunk 2]
```diff
@@ -136,14 +136,24 @@ async function copyTemplateFileAsync(
...
   const templateFilesPath = args.templateFilesPath || path.join(EXPO_DIR, 'template-files');
-  const templatePaths = await new JsonFile(
+  const templatePathsFile = await new JsonFile(
     path.join(templateFilesPath, `${platform}-paths.json`)
-  ).readAsync();
+  ).readAsync() as AndroidPaths;
   const promises: Promise<any>[] = [];
   const skipTemplates: Array<string> = args.skipTemplates || [];
-
+  const templatePaths = {...templatePathsFile.paths, ...templateFilesPath.generateOnly};
```
Comment (sjchmiela) [inline tools/expotools/src/dynamic-macros/generateDynamicMacros.ts]: looks unprettified

**Review by sjchmiela — APPROVED** [review]
LGTM! Thanks! 👍

**Review by tsapeta — APPROVED** [review]

File: tools/expotools/src/dynamic-macros/generateDynamicMacros.ts [hunk 3]
```diff
@@ -136,14 +136,24 @@ async function copyTemplateFileAsync(
...
-  const templatePaths = await new JsonFile(
+  const templatePathsFile = await new JsonFile(
     path.join(templateFilesPath, `${platform}-paths.json`)
-  ).readAsync();
+  ).readAsync() as AndroidPaths;
   const promises: Promise<any>[] = [];
   const skipTemplates: Array<string> = args.skipTemplates || [];
-
+  const templatePaths = { ...templatePathsFile.paths, ...templateFilesPath.generateOnly };
+  console.log(JSON.stringify(templatePaths));
```
Comment (dsokal) [inline tools/expotools/src/dynamic-macros/generateDynamicMacros.ts]:
```suggestion
```

File: tools/expotools/src/dynamic-macros/generateDynamicMacros.ts [hunk 4]
```diff
@@ -136,14 +136,24 @@ async function copyTemplateFileAsync(
   }
 }
 
+type AndroidPaths = {
```
Comment (dsokal) [inline tools/expotools/src/dynamic-macros/generateDynamicMacros.ts]:
```suggestion
interface AndroidPaths {
```

**Comment by IjzerenHein** [comment]
Just for the record, this PR broke generation of `google-services.json` entirely. There was a typo in this line: 

```
const templatePaths = {...templatePathsFile.paths, ...templateFilesPath.generateOnly};
```

which should have been:

```
const templatePaths = { ...templatePathsFile.paths, ...templatePathsFile.generateOnly };
```

However, after fixing it would still throw an error when the `bare-expo` folder did not exist. There was no code that made the copying actually "conditional". I've added the fix and made the copying conditional in 1c9b7a5c547d3b707e690137123dd5e1597c1eaf

**Pull request merged**